samza-test/src/test/java/org/apache/samza/test/processor/IdentityStreamTask.java

```diff
@@ -59,14 +54,7 @@ public void process(
             incomingMessageEnvelope.getMessage()));
     processedMessageCount++;
     if (processedMessageCount == expectedMessageCount) {
-      endLatch.countDown();
+      taskCoordinator.shutdown(TaskCoordinator.RequestScope.ALL_TASKS_IN_CONTAINER);
```

Comment:
awesome change :)

Reply:
Any reason why the `shutdown()` is faster than `latch` in this case?

Reply:
```shutdown()``` is much cleaner because we don't have to share latch between task and test which opens the door for parallel execution. In terms of speed, I believe there isn't big difference.